mon: allow changing hostNetwork settings (#12369)
rook/rook: Documentation/Storage-Configuration/Advanced/ceph-mon-health.md

    osd: 3 osds: 3 up (since 10m), 3 in (since 10m)
[...]
```

> ## Automatic Monitor Failover

Reviewer: nit: add a blank line after the title

> User can enable or disable the `spec.Network.HostNetwork` settings on a running cluster. In such a scenario, mgr and OSD daemons are automatically restarted to use the correct network settings. Rook fails over all the monitors to either use hostNetwork or service IPs based on the network settings provided by the user.

Reviewer: How about this?
```suggestion
Rook will automatically failover the mons when the following settings are updated in the CephCluster CR:
- `spec.network.hostNetwork`: When enabled or disabled, rook fails over all the monitors to either enable or disable host networking
- `spec.network.multiClusterService`: When enabled or disabled, the mons are failed over to start (or stop) using service IPs compatible with the multi-cluster service.
```

    osd: 3 osds: 3 up (since 10m), 3 in (since 10m)
[...]
```

## Automatic Monitor Failover

> Rook will automatically failover the mons when the following settings are updated in the CephCluster CR:

Reviewer 2: ```suggestion
Rook will automatically fail over the mons when the following settings are updated in the CephCluster CR:
```

> - `spec.network.hostNetwork`: When enabled or disabled, rook fails over all the monitors to either enable or disable host networking.

Reviewer 3: In the cluster.yaml please update `hostNetwork` spec

Author: I don't see the `hostNetwork` settings in the cluster.yaml file.  
I can only see ` # enable host networking` comment. Maybe the setting got removed by mistake.

Reviewer: The preferred setting now is `network.provider: host`. The setting `network.hostNetwork: true` is an older way that is still supported for backward compatibility, but we don't need to document it.

> - `spec.network.hostNetwork`: When enabled or disabled, rook fails over all the monitors to either enable or disable host networking.
> - `spec.network.multiClusterService`: When enabled or disabled, the mons are failed over to start (or stop) using service IPs compatible with the multi-cluster service.

Reviewer 2: Some minor suggestions to reduce ambiguity of the wording 

```suggestion
- `spec.network.hostNetwork`: When enabled or disabled, Rook fails over all monitors, configuring them to enable or disable host networking.
- `spec.network.multiClusterService`: When enabled or disabled, Rook fails over all monitors, configuring them to start (or stop) using service IPs compatible with the multi-cluster service.
```
